File: AD-HOC/2138/main.py
from sys import stdin


def most_frequent(n):
    frequency = {}
    for digit in n:
        if digit not in frequency:
            frequency[digit] = 0
        frequency[digit] += 1
    return sorted(frequency.items(), key=lambda digit: [digit[1], digit[0]], reverse=True)[0][0]


for n in stdin:
    print(most_frequent(n))

File: AD-HOC/2138/main.js
function mostFrequent(n) {
    let frequency = {};
    for (let digit of n) {
        if (!(digit in frequency)) frequency[digit] = 0;
        frequency[digit]++;
    }
    return Object.entries(frequency).sort((a, b) => b[1] - a[1] || b[0] - a[0])[0][0];
}

let numbers = require('fs').readFileSync('/dev/stdin', 'utf-8').trim().split('\n');
for (let n of numbers) {
    console.log(mostFrequent(n));
}
